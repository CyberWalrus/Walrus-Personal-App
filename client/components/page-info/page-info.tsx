import { User } from "@client/type/data";
import { iconNames } from "@config/constants";
import * as React from "react";
import { PureComponent, ReactElement } from "react";
import { connect } from "react-redux";
import { Operation } from "../../store/data/data";
import { getUsers } from "../../store/data/selectors";
import { StateApp, ThunkDispatch } from "../../type/reducer";
import Footer from "../footer/footer";
import Header from "../header/header";

interface PropsState {
  users: User[];
}
interface PropsDispatch {
  onUserLoad: () => void;
}
type Props = PropsDispatch & PropsState;
class PageInfo extends PureComponent<Props> {
  public componentWillMount(): void {
    this.props.onUserLoad();
  }
  public render(): ReactElement {
    const users = this.props.users;
    return (
      <div className={`page`}>
        <Header />
        <main className={`page-content`}>
          <section>
            <h2>Info</h2>
            <article>
              {users &&
                users.map((item: User, index: number) => (
                  <p key={index}>
                    {item.login} {item.id}
                  </p>
                ))}
            </article>
            <article>
              <dl className={`info-icon`}>
                {iconNames.map((item: string, index: number) => (
                  <React.Fragment key={index}>
                    <dt>{item} :</dt>
                    <dd className={`icon`}>{item}</dd>
                  </React.Fragment>
                ))}
              </dl>
            </article>
            <article className={`test`}/>
          </section>
        </main>
        <Footer />
      </div>
    );
  }
}

const mapStateToProps = (state: StateApp, ownProps: Props): Props => ({
  ...ownProps,
  users: getUsers(state),
});

const mapDispatchToProps = (dispatch: ThunkDispatch): PropsDispatch => ({
  onUserLoad: (): void => {
    dispatch(Operation.getUsers());
  },
});

export { PageInfo };
export default connect<Props, PropsDispatch, {}, StateApp>(
  mapStateToProps,
  mapDispatchToProps,
)(PageInfo);
