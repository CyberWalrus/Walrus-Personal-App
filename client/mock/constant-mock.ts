import { FormType } from "../../config/constants";
import { Option } from "../hocs/with-form-state/with-form-state";
import { InputCustomInterface } from "../type/component";
import { User, UserRole } from "../type/data";

// tslint:disable-next-line:no-empty
const _functVoid = (): void => {};

const FORM_TYPE: FormType = FormType.SIGN_IN;

const user: User = {
  id: `test`,
  email: `test`,
  isActive: true,
  nickName: `test`,
  login: `test`,
  firstName: `test`,
  lastName: `test`,
  userRoleId: [`test`],
  signUpDate: 1234567,
  created: 1234567,
};
const userRole: UserRole = {
  id: `test`,
  name: `test`,
  isActive: true,
};
const optionForm: Option = {
  values: {},
  formErrors: {},
  formValid: true,
  onChangeUserInput: _functVoid,
  onClickSubmit: _functVoid,
};
const optionInput: InputCustomInterface = {
  formTypes: [FormType.SIGN_UP],
  type: `text`,
  name: `login`,
  placeHolder: `Login`,
  hidenValue: `Login`,
};

const usersEmpty: User[] = [];
const users: User[] = [user, user];

const userRolesEmpty: UserRole[] = [];
const userRoles: UserRole[] = [userRole, userRole];

export {
  user,
  usersEmpty,
  users,
  optionForm,
  FORM_TYPE,
  optionInput,
  userRole,
  userRolesEmpty,
  userRoles,
};
