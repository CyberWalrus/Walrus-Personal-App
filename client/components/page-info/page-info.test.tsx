import * as Enzyme from "enzyme";
import { shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import toJson, { Json } from "enzyme-to-json";
import * as React from "react";
import { users, usersEmpty } from "../../mock/constant-mock";
import { PageInfo } from "./page-info";

Enzyme.configure({ adapter: new Adapter() });
it(`PageInfo correctly render without data`, (): void => {
  const jestFunc = jest.fn();
  const tree: Json = toJson(
    shallow(<PageInfo onUserLoad={jestFunc} users={usersEmpty} />),
  );
  expect(tree).toMatchSnapshot();
});
it(`PageInfo correctly render with data`, (): void => {
  const jestFunc = jest.fn();
  const tree: Json = toJson(
    shallow(<PageInfo onUserLoad={jestFunc} users={users} />),
  );
  expect(tree).toMatchSnapshot();
});
