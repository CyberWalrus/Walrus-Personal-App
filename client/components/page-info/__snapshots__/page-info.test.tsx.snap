// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageInfo correctly render with data 1`] = `
<div
  className="page"
>
  <Header />
  <main
    className="page-content"
  >
    <section>
      <h2>
        Info
      </h2>
      <article>
        <p
          key="0"
        >
          test
           
          test
        </p>
        <p
          key="1"
        >
          test
           
          test
        </p>
      </article>
      <article>
        <dl
          className="info-icon"
        >
          <dt>
            A
             :
          </dt>
          <dd
            className="icon"
          >
            A
          </dd>
          <dt>
            C
             :
          </dt>
          <dd
            className="icon"
          >
            C
          </dd>
          <dt>
            D
             :
          </dt>
          <dd
            className="icon"
          >
            D
          </dd>
          <dt>
            E
             :
          </dt>
          <dd
            className="icon"
          >
            E
          </dd>
          <dt>
            F
             :
          </dt>
          <dd
            className="icon"
          >
            F
          </dd>
          <dt>
            G
             :
          </dt>
          <dd
            className="icon"
          >
            G
          </dd>
          <dt>
            H
             :
          </dt>
          <dd
            className="icon"
          >
            H
          </dd>
          <dt>
            I
             :
          </dt>
          <dd
            className="icon"
          >
            I
          </dd>
          <dt>
            J
             :
          </dt>
          <dd
            className="icon"
          >
            J
          </dd>
          <dt>
            K
             :
          </dt>
          <dd
            className="icon"
          >
            K
          </dd>
          <dt>
            L
             :
          </dt>
          <dd
            className="icon"
          >
            L
          </dd>
          <dt>
            M
             :
          </dt>
          <dd
            className="icon"
          >
            M
          </dd>
          <dt>
            N
             :
          </dt>
          <dd
            className="icon"
          >
            N
          </dd>
          <dt>
            O
             :
          </dt>
          <dd
            className="icon"
          >
            O
          </dd>
          <dt>
            P
             :
          </dt>
          <dd
            className="icon"
          >
            P
          </dd>
          <dt>
            Q
             :
          </dt>
          <dd
            className="icon"
          >
            Q
          </dd>
          <dt>
            R
             :
          </dt>
          <dd
            className="icon"
          >
            R
          </dd>
          <dt>
            S
             :
          </dt>
          <dd
            className="icon"
          >
            S
          </dd>
          <dt>
            T
             :
          </dt>
          <dd
            className="icon"
          >
            T
          </dd>
          <dt>
            U
             :
          </dt>
          <dd
            className="icon"
          >
            U
          </dd>
          <dt>
            V
             :
          </dt>
          <dd
            className="icon"
          >
            V
          </dd>
          <dt>
            W
             :
          </dt>
          <dd
            className="icon"
          >
            W
          </dd>
          <dt>
            X
             :
          </dt>
          <dd
            className="icon"
          >
            X
          </dd>
          <dt>
            Y
             :
          </dt>
          <dd
            className="icon"
          >
            Y
          </dd>
          <dt>
            b
             :
          </dt>
          <dd
            className="icon"
          >
            b
          </dd>
          <dt>
            c
             :
          </dt>
          <dd
            className="icon"
          >
            c
          </dd>
          <dt>
            e
             :
          </dt>
          <dd
            className="icon"
          >
            e
          </dd>
          <dt>
            f
             :
          </dt>
          <dd
            className="icon"
          >
            f
          </dd>
          <dt>
            i
             :
          </dt>
          <dd
            className="icon"
          >
            i
          </dd>
          <dt>
            k
             :
          </dt>
          <dd
            className="icon"
          >
            k
          </dd>
          <dt>
            l
             :
          </dt>
          <dd
            className="icon"
          >
            l
          </dd>
          <dt>
            n
             :
          </dt>
          <dd
            className="icon"
          >
            n
          </dd>
          <dt>
            p
             :
          </dt>
          <dd
            className="icon"
          >
            p
          </dd>
          <dt>
            s
             :
          </dt>
          <dd
            className="icon"
          >
            s
          </dd>
          <dt>
            t
             :
          </dt>
          <dd
            className="icon"
          >
            t
          </dd>
        </dl>
      </article>
      <article
        className="test"
      />
    </section>
  </main>
  <Footer />
</div>
`;

exports[`PageInfo correctly render without data 1`] = `
<div
  className="page"
>
  <Header />
  <main
    className="page-content"
  >
    <section>
      <h2>
        Info
      </h2>
      <article />
      <article>
        <dl
          className="info-icon"
        >
          <dt>
            A
             :
          </dt>
          <dd
            className="icon"
          >
            A
          </dd>
          <dt>
            C
             :
          </dt>
          <dd
            className="icon"
          >
            C
          </dd>
          <dt>
            D
             :
          </dt>
          <dd
            className="icon"
          >
            D
          </dd>
          <dt>
            E
             :
          </dt>
          <dd
            className="icon"
          >
            E
          </dd>
          <dt>
            F
             :
          </dt>
          <dd
            className="icon"
          >
            F
          </dd>
          <dt>
            G
             :
          </dt>
          <dd
            className="icon"
          >
            G
          </dd>
          <dt>
            H
             :
          </dt>
          <dd
            className="icon"
          >
            H
          </dd>
          <dt>
            I
             :
          </dt>
          <dd
            className="icon"
          >
            I
          </dd>
          <dt>
            J
             :
          </dt>
          <dd
            className="icon"
          >
            J
          </dd>
          <dt>
            K
             :
          </dt>
          <dd
            className="icon"
          >
            K
          </dd>
          <dt>
            L
             :
          </dt>
          <dd
            className="icon"
          >
            L
          </dd>
          <dt>
            M
             :
          </dt>
          <dd
            className="icon"
          >
            M
          </dd>
          <dt>
            N
             :
          </dt>
          <dd
            className="icon"
          >
            N
          </dd>
          <dt>
            O
             :
          </dt>
          <dd
            className="icon"
          >
            O
          </dd>
          <dt>
            P
             :
          </dt>
          <dd
            className="icon"
          >
            P
          </dd>
          <dt>
            Q
             :
          </dt>
          <dd
            className="icon"
          >
            Q
          </dd>
          <dt>
            R
             :
          </dt>
          <dd
            className="icon"
          >
            R
          </dd>
          <dt>
            S
             :
          </dt>
          <dd
            className="icon"
          >
            S
          </dd>
          <dt>
            T
             :
          </dt>
          <dd
            className="icon"
          >
            T
          </dd>
          <dt>
            U
             :
          </dt>
          <dd
            className="icon"
          >
            U
          </dd>
          <dt>
            V
             :
          </dt>
          <dd
            className="icon"
          >
            V
          </dd>
          <dt>
            W
             :
          </dt>
          <dd
            className="icon"
          >
            W
          </dd>
          <dt>
            X
             :
          </dt>
          <dd
            className="icon"
          >
            X
          </dd>
          <dt>
            Y
             :
          </dt>
          <dd
            className="icon"
          >
            Y
          </dd>
          <dt>
            b
             :
          </dt>
          <dd
            className="icon"
          >
            b
          </dd>
          <dt>
            c
             :
          </dt>
          <dd
            className="icon"
          >
            c
          </dd>
          <dt>
            e
             :
          </dt>
          <dd
            className="icon"
          >
            e
          </dd>
          <dt>
            f
             :
          </dt>
          <dd
            className="icon"
          >
            f
          </dd>
          <dt>
            i
             :
          </dt>
          <dd
            className="icon"
          >
            i
          </dd>
          <dt>
            k
             :
          </dt>
          <dd
            className="icon"
          >
            k
          </dd>
          <dt>
            l
             :
          </dt>
          <dd
            className="icon"
          >
            l
          </dd>
          <dt>
            n
             :
          </dt>
          <dd
            className="icon"
          >
            n
          </dd>
          <dt>
            p
             :
          </dt>
          <dd
            className="icon"
          >
            p
          </dd>
          <dt>
            s
             :
          </dt>
          <dd
            className="icon"
          >
            s
          </dd>
          <dt>
            t
             :
          </dt>
          <dd
            className="icon"
          >
            t
          </dd>
        </dl>
      </article>
      <article
        className="test"
      />
    </section>
  </main>
  <Footer />
</div>
`;
